Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -1993,6 +1982,10 @@ void AppendPipeline::resetPipeline()
 {
     ASSERT(WTF::isMainThread());
     LOG_MEDIA_MESSAGE("resetting pipeline");
+    m_atLeastABufferLeftAppsrcMutex.lock();
```

Comment:
No, I can't, the rest of the method forces to claim the mutex somewhere else and then we have a deadlock. That was the first thing I tried. I could use unlockEarly() on the MutexLocker, but I think it doesn't pay off for such straight method.